storage/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteLogManager.java

```diff
@@ -293,6 +293,7 @@ public void resizeExpirationThreadPool(int newSize) {
     public void resizeReaderThreadPool(int newSize) {
         int currentSize = remoteStorageReaderThreadPool.getCorePoolSize();
         LOGGER.info("Updating remote reader thread pool size from {} to {}", currentSize, newSize);
+        remoteStorageReaderThreadPool.setMaximumPoolSize(newSize);
```

Comment:
The patch still fails when we try to decrease the thread count. Confirmed it with below unit test:

```java
@Test
void testUpdateRemoteStorageReaderThreads() {
  assertEquals(10, remoteLogManager.readerThreadPoolSize());
  remoteLogManager.resizeReaderThreadPool(6);
  assertEquals(6, remoteLogManager.readerThreadPoolSize());
  remoteLogManager.resizeReaderThreadPool(12);
  assertEquals(12, remoteLogManager.readerThreadPoolSize());
}
```

Reply:
Sorry, I missed the decreasing thread count case. Updated `resizeReaderThreadPool` to handle it. Thanks.